### DISCUSSION TITLE
Issue serializing data attribute in Pay::Subscription

### DISCUSSION THREAD

**Discussion by ferblape** [pay-rails/pay#958] [Q&A]
Hi all,

sorry if it's already been asked, I couldn't find any reference. 

I've setup an integration with Stripe, and has worked fine so far in the last 6 months, but a couple of times we've had an issue in the Pay::Subscription model, where the data attribute is not properly serialized as Ruby Hash, it's serialized as String:

```
 data:
  "subscription_items [{\"id\"=>\"si_xxxxx\", \"price\"=>{\"id\"=>\"price_xxxxx\", \"type\"=>\"recurring\", \"active\"=>true, \"object\"=>\"price\", \"created\"=>1681490013, \"product\"=>\"prod_xxxx\", \"currency\"=>\"eur\", \"livemode\"=>true, \"metadata\"=>{}, \"nickname\"=>nil, \"recurring\"=>{\"interval\"=>\"year\", \"usage_type\"=>\"licensed\", \"interval_count\"=>1, \"aggregate_usage\"=>nil, \"trial_period_days\"=>nil}, \"lookup_key\"=>nil, \"tiers_mode\"=>nil, \"unit_amount\"=>300000, \"tax_behavior\"=>\"exclusive\", \"billing_scheme\"=>\"per_unit\", \"custom_unit_amount\"=>nil, \"transform_quantity\"=>nil, \"unit_amount_decimal\"=>\"300000\"}, \"metadata\"=>{}, \"quantity\"=>1}]",
```

This is the check that I ran after:

```
[20] pry(main)> Pay::Subscription.all.select{ |s| s.data.is_a?(String) }.size
=> 0
```

Reprocessing the attribute and assigning a hash fixes the issue, but I'd like to know if anybody had the same issue before. I think I could try to reproduce it because I have the webhook sent by Stripe.

**Comment by excid3** [comment]
What version of Pay are you using? What database & column type? Any other details?

**Reply by ferblape** [reply]
Pay 6.8.1
